docs/minio-job.md

```diff
@@ -0,0 +1,208 @@
+kind: PolicyBinding
+metadata:
+  name: mc-job-binding
+spec:
+  application:
+    serviceaccount: mc-job-sa
+  policies:
+    - consoleAdmin
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mytestsecret
+data:
+  PASSWORD: cGVkcm8xMjM=
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mytestsecretenvs
+data:
+  USER: ZGFuaWVs
+  PASSWORD: ZGFuaWVsMTIz
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: mytestconfig
+data:
+  policy.json:  |
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "s3:*"
+                ],
+                "Resource": [
+                    "arn:aws:s3:::memes",
+                    "arn:aws:s3:::memes/*"
+                ]
+            }
+        ]
+    }
+---
+apiVersion: job.min.io/v1alpha1
+kind: MinIOJob
+metadata:
+  name: minio-test-job
+spec:
+  mcImage: quay.io/minio/mc:latest # optional, defaults to minio/mc:latest
+  serviceAccountName: mc-job-sa
+  securityContext: {}
+  containerSecurityContext: {}
+  tenant:
+    name: mytest-minio
+  commands:
+    - op: make-bucket
+      args:
+        name: memes
+    - name: add-my-user-1
+      op: admin/user/add
+      args:
+        user: ${USER}
+        password: ${PASSWORD}
+      envFrom:
+        - secretRef:
+            name: mytestsecretenvs
+    - name: add-my-user-2
+      op: admin/user/add
+      args:
+        user: pedro
+        password: $PASSWORD
+      env:
+        - name: PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mytestsecret
+              key: PASSWORD
+    - name: add-my-policy
+      op: admin/policy/create
+      args:
+        name: memes-access
+        policy: /temp/policy.json
+      volumeMounts:
+        - name: policy
+          mountPath: /temp
+      volumes:
+        - name: policy
+          configMap:
+            name: mytestconfig
+            items:
+              - key: policy.json
+                path: policy.json
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: daniel
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: pedro
+    - op: stat
+      command:
+        - "mc"
+        - "stat"
+        - "myminio/memes"
+```
+The MinioJob is a Kubernetes Job that runs mc commands. It uses the MinIO client (mc) to interact with the MinIO server.
+## mcImage
+The `mcImage` field specifies the Docker image that will be used to run the mc commands. In this case, the image is `minio/mc:latest`.
+## serviceAccountName
+The `serviceAccountName` field specifies the name of the Kubernetes ServiceAccount that will be used to run the mc commands. In this case, the ServiceAccount is `mc-job-sa`.
```

Comment:
We might need guidance on how the Kubernetes Service Account gets associated to a user on MinIO right?  I see we have a configmap above to assign a policy to the user, but the connection to MinIO Tenant isn't as clear to me.

Reply:
OK - so configuring and using Operator STS is required here. Any other way to auth or only via Operator STS?

Reply:
@ravindk89 Do you mean that you expect to use it using a manual access-key / secret-key to avoid using STS?

Reply:
@ramondeklein yeah - if it's STS only, then it implies that when we have webdocs for MinioJob we also need to ensure Operator STS is documented and fully validated.